IdentityCore/src/requests/broker/MSIDSSOExtensionSilentTokenRequest.m

```diff
@@ -85,6 +85,17 @@ - (instancetype)initWithRequestParameters:(MSIDRequestParameters *)parameters
 #if TARGET_OS_OSX
             strongSelf.ssoTokenResponseHandler.externalCacheSeeder = strongSelf.externalCacheSeeder;
 #endif
+            // Edge case: both response and error is nil. This happens if Apple SSO return the wrong callback.
+            // For normal scenario (we have response or error from SSO-Ext, keep the same logic)
+            if (!operationResponse && !error)
```

Comment:
So the bug is in MSIDSSOExtensionTokenRequestDelegate, but we are adding fixes in the external module. I think we should keep the contract: the delegate should return either error or response. If both are nil (due to the Apple API bug) we can return an internal error (like MSIDErrorSSOExtensionUnexpectedError). Once we get this error in `completionBlockWrapper ` in MSIDSilentController, we can return the result of the local silent call.

In future, we can reuse this MSIDErrorSSOExtensionUnexpectedError for the fallback logic in other modules. Whenever we get, it will mean the SSO ext doesn't work properly and we want to fallback to the local flows.

Reply:
It's actually a very nice idea. I like that. Will update the code.
Thanks

Reply:
Updated